main.go

```diff
@@ -81,7 +82,16 @@ func main() {
 		os.Exit(exitFailure)
 	}
 
-	shim, err := newShim(agentAddr, container, uint32(pid))
+	// Wait on SIGUSR1 if it is an init process shim, in which case
+	// container ID equals to exec ID.
+	if container == execId {
```

Comment:
I think we should consider moving that case to the agent to simplify the shim and runtime code and logic. See my [comment](https://github.com/kata-containers/shim/issues/18#issuecomment-355517646).